packages/toolpad-studio-components/src/DataGrid.tsx

```diff
@@ -1262,7 +1270,18 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
   }, [nodeRuntime, rows]);
 
   const renderedColumns = React.useMemo<GridColDef[]>(() => {
-    const result = [...columns];
+    const mapEditableToColumns = columns.map((column) => {
+      // Don't touch the existing columns with `editable` value, example: `id` column
+      if (column.editable || column.editable === false) {
+        return column;
+      }
+
+      column.editable = isRowUpdateModelAvailable;
```

Comment:
I think we should avoid mutating items in `.map`. Perhaps something like the following would work?
```tsx
return column.editable === undefined ? { ...column, editable: isRowUpdateModelAvailable } : column;
```